noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -14,50 +14,134 @@ use dep::protocol_types::{
+/// | 1. Constrained or unconstrained encryption                                                             |
+/// | - When the encryption is constrained we have a guarantee that the resulting ciphertext is formed       |
+/// |   correctly. If we don't need this guarantee and we are fine with trusting the sender, it is           |
+/// |   recommended to use unconstrained encryption as it's cheaper constraint-wise.                         |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 2. Delivery mechanism                                                                                  |
+/// | - The message ciphertext can be delivered either via the private log stream or as an offchain message. |
+/// | - The private log stream is handled by the Aztec protocol, is submitted to L1 blobs and consumes DA.   |
+/// | - The offchain message mechanism is implemented by the app and hence is not expected to consume DA     |
+/// |   (imagine an app submitting all the ciphertext to cloud storage).                                     |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 3. Constrained or unconstrained tagging                                                                |
+/// | - Tagging is an indexing mechanism that enables recipients to locate private logs in the system. When  |
+/// |   tagging is unconstrained, there is not a cryptographic guarantee about it being performed correctly. |
+/// |   This may prevent recipients from successfully discovering their messages. Therefore, unconstrained   |
+/// |   tagging should only be used in scenarios where the recipient is trusted or has clear incentives to   |
+/// |   ensure proper message delivery.                                                                      |
+/// | - Tagging is not used in the offchain message mechanism.                                               |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+///
+/// # Message authenticity
+/// Note that we always commit to the plaintext and the recipient is therefore always able to verify the authenticity
+/// of the message no matter the delivery mechanism.
+///
+///
+/// # Message Delivery Variants and Their Use Cases
+///
+/// The following section outlines the recommended delivery variants and when to use each one.
+/// Each variant represents a different tradeoff between security guarantees, gas costs, and trust assumptions.
+/// - We are currently still exploring the design space but it seems that there are only 2 meaningful combinations:
```

Comment:
I wonder if the reviewer agrees with this. I feel like we are converging only on 2 meaningful configurations here.

I can imagine that a dev might want to use unconstrained encryption, unconstrained tagging and private log stream if he wants it cheap but is lazy to implement the offchain message delivery in the app. However I think in that case we should spin up some centralized service such that this is simple to do and headache-free for devs. If we do this then there really doesn't seem to be any reason for existence of other combinations.